src/charm.py

```diff
@@ -30,6 +31,20 @@ def __init__(self, *args):
         super().__init__(*args, distro=OpenSearchTarball)  # OpenSearchSnap
 
         self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.leader_elected, self._on_leader_elected)
```

Comment:
These are already on the `opensearch_base_charm.py` file in the `OpenSearchBaseCharm` class. 
Basically anything common to the VM and K8s charm, should be in that class. Only the things that are proper to the VM charm should be in the `charm.py` file